explicit send() arguments

gronke opened the pull request:
Explicitly names arguments of the `send()` methods. This is particularly useful when using strong typing and working with the Python CLI.

igalic commented on libzfs.pyi:
```diff
@@ -0,0 +1,345 @@
+    to_native = ... # type: Any
+    to_property = ... # type: Any
+
+class datetime(date):
```
this can be fixed with
```python
from datetime import date
```

araujobsd commented:
What is the purpose?

igalic commented:
being able to [mypy](http://mypy.readthedocs.io/en/latest/) test a project that uses py-libzfs

araujobsd commented:
It is an experimental project and very much intrusive, I would prefer see pytest covering libzfs rather than add an additional layer of complexity provided by an experimental project.

Maybe when it gets mature and other big projects adopt it we can consider, but for now, it is just a big NO.

Thanks for the PR anyway.

gronke committed:
explicit send() arguments

william-gr commented on libzfs.pyx:
```diff
@@ -2401,19 +2401,16 @@ cdef class ZFSDataset(ZFSObject):
             i.umount_recursive(force)
 
-    def send(self, fd, **kwargs):
+    def send(self, fd, fromname=None, toname=None, flags=set()):
```
set() is not a singletone, should not be used as argument:

```
>>> def test(foo, flags=set()):
...     flags.add(foo)
...     print(flags)
... 
>>> test('foo')
{'foo'}
>>> test('bar')
{'bar', 'foo'}
```

gronke replied:
Wow, that's amazing! That makes so much sense because the set() does not get created when calling the function. Great catch!

gronke committed:
make default flags argument a non-singleton

addresses https://github.com/freenas/py-libzfs/pull/4#pullrequestreview-110777191
